### ISSUE TITLE
aws_lambda_function CRASH if "publish" argument is integer instead of boolean

### ISSUE THREAD

**Issue by gabriel-soltz-olx** [hashicorp/terraform#20591]
### Terraform Version
```
15:51 $ terraform --version
Terraform v0.11.11
+ provider.archive v1.1.0
+ provider.aws v1.60.0
+ provider.external v1.0.0
+ provider.null v2.0.0
+ provider.template v2.0.0
```

### Terraform Configuration Files
```hcl
resource "aws_lambda_function" "test_lambda" {
  description      = "AWS Security Monitoring 3.0"
  filename         = "${data.archive_file.lambda_zip.output_path}"
  runtime          = "python2.7"
  timeout          = 60
  memory_size      = 256
  publish          = 256
  function_name    = "${var.lambda_function_name}"
  role             = "${aws_iam_role.iam_role_lambda_monitoring.arn}"
  handler          = "lambda_event.lambda_event"
  source_code_hash = "${data.archive_file.lambda_monitoring_zip.output_base64sha256}"
}
```

### Crash Output
```

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

### Expected Behavior
Error related to publish argument...

### Actual Behavior
CRASH

### Steps to Reproduce
1. Define a lambda with an integer instead a boolean in publish argument. 
2. make plan

**Comment by apparentlymart** [comment]
Hi @gabriel-soltz-olx! Sorry for this crash and thanks for reporting it.

In the forthcoming Terraform v0.12.0 release Terraform Core pre-validates that all arguments match the types expected by the provider, whereas in v0.11 and earlier it is the provider's responsibility to validate its own configuration. Therefore I expect that this is already inadvertently fixed in [v0.12.0-beta1](https://www.hashicorp.com/blog/announcing-terraform-0-1-2-beta1) due to that extra check, and that fix will also be included in the final release.

Since this change has already been made, I'm going to put this in the v0.12.0 milestone just to prompt us to verify that it is indeed fixed to return a proper type checking error.

**Comment by mildwonkey** [comment]
I can confirm that this is fixed in master!

```
Error: Incorrect attribute value type

  on aws.tf line 28, in resource "aws_lambda_function" "test_lambda":
  28:   publish          = 256

Inappropriate value for attribute "publish": bool required.
```
